packages/calling-stateful-client/src/CallSubscriber.ts

```diff
@@ -167,11 +167,11 @@ export class CallSubscriber {
 
   public unsubscribe = (): void => {
     this._call.off('stateChanged', this.stateChanged);
-    this._call.off('stateChanged', this.initCaptionSubscriber);
+    this._call.off('stateChanged', () => {this._safeSubscribe(this.initCaptionSubscriber)});
```

Comment:
I believe because this creates a new anonymous function the `off` won't match the `on` so we won't successfully unsubscribe.

Thinking about how best to create the helper wrapper...

Reply:
hmmm good catch. 
One way to mitigate it is by enforcing safeSubscribe use in the listeners itself.

Reply:
Instead, i decided to wrap everything into a function const. This works. Tested and acting appropriately.